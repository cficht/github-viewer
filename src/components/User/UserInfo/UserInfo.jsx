import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ login, followers, following, html_url }) => (
  <>
    <p>{login}</p>
    <p>{followers}</p>
    <p>{following}</p>
    <p>{html_url}</p>
  </>
);

UserInfo.propTypes = {
  login: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired
};

export default UserInfo;
