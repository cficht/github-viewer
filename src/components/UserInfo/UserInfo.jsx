import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ login, followers, following, html_url }) => (
  <>
    <h2>Username: {login}</h2>
    <p>Followers: {followers}</p>
    <p>Following: {following}</p>
    <p>URL: {html_url}</p>
  </>
);

UserInfo.propTypes = {
  login: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired
};

export default UserInfo;
