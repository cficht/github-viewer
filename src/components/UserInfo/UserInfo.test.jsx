import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from './UserInfo';

describe('UserInfo component', () => {
  it('renders UserInfo', () => {
    const wrapper = shallow(<UserInfo 
      login="cficht"
      followers={2}
      following={7}
      html_url="https://github.com/cficht"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
