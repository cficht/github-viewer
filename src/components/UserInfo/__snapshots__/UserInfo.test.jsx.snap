// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo component renders UserInfo 1`] = `
<Fragment>
  <p>
    cficht
  </p>
  <p>
    2
  </p>
  <p>
    7
  </p>
  <p>
    https://github.com/cficht
  </p>
</Fragment>
`;
